### ISSUE TITLE
Android video file upload

### ISSUE THREAD

**Issue by arianaa30** [ThibaultBee/srtdroid#19]
I'm new to SRT. Looks very promising protocol. I have a video delivery app on Android that captures video from camera, then uploads the video file to the server and another device can use it for playback later. 

Will use of this protocol help with faster uploads/downloads? And do we have an example Android app and server for this purpose that I can use?

**Comment by ThibaultBee** [comment]
Hi,

For an upload SRT might be faster. It depends of your upload protocol. As SRT is a UDP-based protocol, it will be faster by design than most TCP-based protocol. Your question is SRT specific, for this type of question you should directly reach SRT developers on their [github](https://github.com/Haivision/srt) or slack.

Anyway, it shouldn't be difficult to create quickly an Android app (all you need is [`Socket.sendFile`](https://thibaultbee.github.io/srtdroid/dokka/lib/com.github.thibaultbee.srtdroid.models/-socket/send-file.html). For the server, you should look at [SRT examples](https://github.com/Haivision/srt/blob/master/examples/recvfile.cpp).

You should know the main use case of SRT is livestreaming. For a livestreaming Android app, you can have a look at my other project [StreamPack](https://github.com/ThibaultBee/StreamPack).

**Comment by arianaa30** [comment]
Thanks. So in my usecase, I should deal with it like a file transfer (which still is faster). It doesn't matter if the file is actually a mp4 video file or any other file.

**Comment by ThibaultBee** [comment]
> Thanks. So in my usecase, I should deal with it like a file transfer (which still is faster). It doesn't matter if the file is actually a mp4 video file or any other file. Is it right?
No, it doesn't matter. You can send any type of file with `sendFile`.

> And second question: I'm thinking to chunk up the video file into 5s chunks and send them in parallel. Do you think it will help with faster uploads?
Not sure it will be faster as the limit here is the bandwidth. If you send one file or multiple files, you are still limited by the same bandwidth (it won't increase your bandwidth). The advantage of chunking video file into multiple chunk is if you want to stop/restart to send the file, you don't have to send the whole file each time, you have to send only the chunk that wasn't received. But this is something you will have to implement on your own.

**Comment by arianaa30** [comment]
I see. So wondering, how do download accelerators cause faster downloads by creating multiple parallel threads?

**Comment by ThibaultBee** [comment]
This is getting far from SRT. Apparently, splitting in multiples files
could help on slow connections. See
https://stackoverflow.com/questions/93642/how-do-download-accelerators-work
